test/unit/math/mix/meta/eigen_utils.hpp

```diff
@@ -0,0 +1,298 @@
+ *   `EigenBase` types.
+ * @tparam eigen_dense_v logical for whether `Checker` should pass for
+ *   `DenseBase` types.
+ * @tparam eigen_matrix_v logical for whether `Checker` should pass for
+ *   `EigenMatrixBase` types.
+ * @tparam eigen_matrix_expr_v logical for whether `Checker` should pass for
+ *   `EigenMatrixBase` expression types.
+ * @tparam eigen_map_v logical for whether `Checker` should pass for
+ *   `Map` expression types.
+ * @tparam Checker A struct taking in an expression template type and returning
+ *  a true if check is satisfied and false otherwise.
+ * @tparam EigenDims template parameter pack of `int` defining matrix
+ *  compile time dimensions.
+ */
+template <bool eigen_base_v, bool eigen_dense_v, bool eigen_matrix_v,
+          bool eigen_matrix_expr_v, bool eigen_map_v,
+          template <class...> class Checker, int... EigenDims>
+void all_eigen_dense_matrix() {
+  using stan::math::fvar;
+  using stan::math::var;
+
+  test_eigen_dense<eigen_base_v, eigen_dense_v, eigen_matrix_v,
+                   eigen_matrix_expr_v, eigen_map_v, Checker, Eigen::Matrix,
+                   double, EigenDims...>();
+  test_eigen_dense<eigen_base_v, eigen_dense_v, eigen_matrix_v,
+                   eigen_matrix_expr_v, eigen_map_v, Checker, Eigen::Matrix,
+                   var, EigenDims...>();
+  test_eigen_dense<eigen_base_v, eigen_dense_v, eigen_matrix_v,
+                   eigen_matrix_expr_v, eigen_map_v, Checker, Eigen::Matrix,
+                   fvar<double>, EigenDims...>();
+  test_eigen_dense<eigen_base_v, eigen_dense_v, eigen_matrix_v,
+                   eigen_matrix_expr_v, eigen_map_v, Checker, Eigen::Matrix,
+                   fvar<var>, EigenDims...>();
+  test_eigen_dense<eigen_base_v, eigen_dense_v, eigen_matrix_v,
+                   eigen_matrix_expr_v, eigen_map_v, Checker, Eigen::Matrix,
+                   fvar<fvar<double>>, EigenDims...>();
+}
+
+/**
+ * Test type checking for dense arrays with all stan scalar types.
+ * @tparam eigen_base_v logical for whether `Checker` should pass for
+ *   `EigenBase` types.
+ * @tparam eigen_dense_v logical for whether `Checker` should pass for
+ *   `DenseBase` types.
+ * @tparam eigen_matrix_v logical for whether `Checker` should pass for
+ *   `EigenMatrixBase` types.
+ * @tparam eigen_matrix_expr_v logical for whether `Checker` should pass for
+ *   `EigenMatrixBase` expression types.
+ * @tparam eigen_map_v logical for whether `Checker` should pass for
+ *   `Map` expression types.
+ * @tparam Checker A struct taking in an expression template type and returning
+ *  a true if check is satisfied and false otherwise.
+ * @tparam EigenDims template parameter pack of `int` defining matrix
+ *  compile time dimensions.
+ */
+template <bool eigen_base_v, bool eigen_dense_v, bool eigen_array_v,
+          bool eigen_array_expr_v, bool eigen_map_v,
+          template <class...> class Checker, int... EigenDims>
+void all_eigen_dense_array() {
```

Comment:
Can you unify the formatting between this and previous function? Differences in blank lines make it less obvious they are almost the same.